modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/service/impl/PLOEntryLocalServiceImpl.java

```diff
@@ -118,6 +127,33 @@ public int getPLOEntriesCount(long companyId) {
 		return ploEntryPersistence.countByCompanyId(companyId);
 	}
 
+	@Override
+	public void importPLOEntries(
+			long companyId, InputStream inputStream, String languageId,
+			long userId)
+		throws IOException, PortalException {
+
+		Properties properties = new Properties();
+
+		properties.load(
+			new InputStreamReader(inputStream, StandardCharsets.UTF_8));
+
+		if (properties.isEmpty()) {
+			throw new InvalidPropertiesFileException("Invalid properties file");
+		}
+
+		Enumeration<String> enumeration =
+			(Enumeration<String>)properties.propertyNames();
+
+		while (enumeration.hasMoreElements()) {
+			String key = enumeration.nextElement();
+
+			addOrUpdatePLOEntry(
+				companyId, userId, key, languageId,
+				properties.getProperty(key));
```

Comment:
I believe that the previous logic would catch an exception and add the error to session errors. We could do the same here by catching errors inside the loop, adding the message to a list, and throwing our own exception after the loop if the list is not empty. That would be _similar_ to how it was before.

Reply:
hmm wouldn't that require having the `actionRequest` as a method parameter?

Reply:
I don't remember seeing such pattern. And my quick search now didn't show up anything useful (or maybe idk what I'm looking for).

Do you have an example? I do not know how to collect the errors in a "aggregate Exception".

Reply:
Similar to this: https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-language/portal-language-override-web/src/main/java/com/liferay/portal/language/override/web/internal/portlet/action/ImportTranslationsMVCActionCommand.java?plain=1#L104-L111

Reply:
That's how we used to do it. For this case, really we just need to construct a string that will be used as a message, then throw an exception with that message. That's all I mean by "aggregate".

Reply:
So, for each error that might be thrown, we add it to a String and then throws it at the end?

```java
StringBuilder errors = new StringBuilder();

while (enumeration.hasMoreElements()) {
    String key = enumeration.nextElement();
    try {
	addOrUpdatePLOEntry(
		companyId, userId, key, languageId,
		properties.getProperty(key));
    } catch(Exception exception) {
        errors.append("Could not save key ");
        errors.append(key);
    }
}

if(errors.isEmpty()) {
    throw new SomeExceptionIHaveToCreate(errors);
}
```

Reply:
Yup! Just remember to flip the `if` condition 😉

Reply:
This of course is all to preserve the original behavior as much as possible.

It is also a legitimate option to fail the whole import if there is an error, but we should make that kind of change on purpose instead of accidentally.

Reply:
@ethib137  The current behavior on master is to add the exceptions for the `SessionErrors`. If one key fails with empty value, another with empty key and another invalid language id, the user will see three error messages in the UI.

But, the keys that didn't have any issues will be added to the database.

The code I made in this PR stops at the first error, rollback any PLOEntry that might have been added (this is automatic), and then shows the error message in the UI. For me it makes more sense that if you say an error occurred while importing a file, nothing in the database has changed 🧐. 

The drawback is that the user wouldn't see more than one error at the same time. If we go with what Drew proposed above the user will see multiple errors in the UI, but we would be spending time adding valid keys, just to realize they will be rolled back when the exception is thrown.

I'll consider this as a product decision, do we keep the current behavior or go with the one that stops at the first error it encounters?

Reply:
Oh damn it, I forgot. Evan is on PTO for a week. @drewbrokke, do you see any issue with the current approach? That is, stopping at the first error encountered and not saving any previously added entries